src/catalog.h

```diff
@@ -644,58 +657,73 @@ typedef enum CacheType
 	_MAX_CACHE_TYPES
 } CacheType;
 
-typedef struct Catalog
+typedef enum CatalogType CatalogType;
+
+typedef struct CatalogTableInfo
+{
+	const char *schema_name;
+	const char *name;
+	Oid			id;
+	Oid			serial_relid;
+	Oid			index_ids[_MAX_TABLE_INDEXES];
+} CatalogTableInfo;
+
+typedef struct CatalogBase
 {
 	char		database_name[NAMEDATALEN];
 	Oid			database_id;
 	Oid			schema_id;
-	struct
-	{
-		const char *schema_name;
-		const char *name;
-		Oid			id;
-		Oid			serial_relid;
-		Oid			index_ids[_MAX_TABLE_INDEXES];
-	}			tables[_MAX_CATALOG_TABLES];
+	Oid			owner_uid;
+} CatalogBase;
+
+typedef struct Catalog
+{
+	CatalogBase base;
```

Comment:
Hey I am no longer sure that CatalogBase really needs to be embedded in Catalog. Won't each catalog then have the same exact CatalogBase values. Can't we make it it's own independent object?

Reply:
Alternatively can we make it part of catalog but just called SystemInfo? And then it shouldn't be part of other catalogs. I guess what confuses me is why would this be a base or prefix for other catalog?

Reply:
Renamed to `CatalogDatabaseInfo` and removed from within the `Catalog` struct.